Orange/widgets/data/owselectcolumns.py

```diff
@@ -97,7 +102,76 @@ def dropMimeData(self, mime, action, row, column, parent):
         if row < 0:
             row = self.rowCount()
 
+        if self.primitive and not all(var.is_primitive() for var in variables):
+            variables = [var for var in variables if var.is_primitive()]
+            self[row:row] = variables
+            mime.setProperty("_moved", variables)
```

Comment:
@ales-erjavec, MIME is passed from source of the drag to its destination. In this PR, destination may accept only some data, so it must have a way to pass the list of accepted variables back to the source. `drag.exec` does not return values. Here I treated mime data as a shared, two-way object. It works, but is it OK? Documentation neither mentions nor prohibits it.

Reply:
Probably fine.